core/lib/zksync_core/src/state_keeper/io/common.rs

```diff
@@ -75,13 +23,18 @@ pub(crate) async fn load_pending_batch(
     validation_computational_gas_limit: u32,
     chain_id: L2ChainId,
 ) -> Option<PendingBatchData> {
-    let (system_env, l1_batch_env) = load_l1_batch_params(
-        storage,
+    let previous_l1_batch_hash = wait_for_prev_l1_batch_params(storage, current_l1_batch_number)
```

Comment:
IMO, this construction is unnecessarily brittle; nowhere does it guarantee that the hash supplied to `with_prev_batch_hash` won't be mismatched (e.g., by loading a hash for a different L1 batch).

Reply:
Yeah, but sometimes it's easier to get it somewhere outside. 
And it's users responsibility to get it correct